### PR TITLE
Updated the rename column to use the new implementations for a dialog

### PR DESCRIPTION
**Pull request by deaspo**
Updated the rename column to use the new implementations for a dialog

### CODE REVIEW

File: instat/dlgName.vb [hunk 1]
```diff
@@ -16,53 +16,61 @@
...
-        ucrReceiverNames.Enabled = True
-        ucrReceiverNames.Visible = True
-        bUseDefaultName = True
-        'txtName.Text = ""
+        txtName.Text = ""
+        ucrReceiverName.SetMeAsReceiver()
     End Sub
 
     Private Sub ucrBase_ClickReset(sender As Object, e As EventArgs) Handles ucrBase.ClickReset
+        ucrSelectVariables.Reset()
```
Comment (dannyparsons) [inline instat/dlgName.vb]: This is part of the 'defaults' so it can go in DefaultSettings() instead of being separate for Reset.

File: instat/dlgName.vb [hunk 2]
```diff
@@ -16,53 +16,61 @@
...
-        ucrReceiverNames.txtReceiverSingle.Text = frmEditor.grdData.CurrentWorksheet.ColumnHeaders(frmEditor.grdData.CurrentWorksheet.SelectionRange.Col).[Text]
+        ucrReceiverName.Selector = ucrSelectVariables
+        If bFirstLoad Then
+            defaultSettings()
+            bFirstLoad = False
+        End If
+        TestOKEnabled()
     End Sub
 
     Private Sub defaultSettings()
```
Comment (dannyparsons) [inline instat/dlgName.vb]: Capital D needed for defaults (or change to SetDefaults for consistency).

**Comment by deaspo** [comment]
I have updated the changes

**Comment by dannyparsons** [comment]
I've discussed with David. We like what you've done in frmEditor to set options before opening the dialog.

The way to make this work nicely is to merge `setSettings` and `SetDefaults` as one `Sub`. This would have two optional parameters, one for Dataframe name and one for column name. The defaults will probably be empty strings for both. Then you still just have `SetDefaults` that is used just like in other dialogs, except yours has some optional parameters to do what your `setSettings` currently does.

Because there will be optional parameters, you can then remove your extra line in frmMain.

There are a few other small corrections to made that I will comment on the specific lines.

Let me know if you want to discuss this further.

File: instat/dlgName.vb [hunk 3]
```diff
@@ -16,53 +16,61 @@
...
+        Else
+            ucrBase.OKEnabled(False)
+        End If
+    End Sub
 
-    Private Sub ucrSingle_Leave(sender As Object, e As EventArgs) Handles ucrReceiverNames.Leave
+    Private Sub ucrSelectVariables_DataFrameChanged() Handles ucrSelectVariables.DataFrameChanged
         ucrBase.clsRsyntax.AddParameter("data_name", Chr(34) & frmEditor.grdData.CurrentWorksheet.Name & Chr(34))
-        ucrBase.clsRsyntax.AddParameter("column_name", Chr(34) & frmEditor.grdData.CurrentWorksheet.ColumnHeaders(frmEditor.grdData.CurrentWorksheet.SelectionRange.Col).[Text] & Chr(34))
+        TestOKEnabled()
```
Comment (dannyparsons) [inline instat/dlgName.vb]: Since only the receivers are part of the `TestOKEnabled`, you don't actually need to do `TestOKEnabled` here.

File: instat/dlgName.vb [hunk 4]
```diff
@@ -16,53 +16,61 @@
...
+        If Not ucrReceiverName.IsEmpty() And txtName.Text <> "" Then
+            ucrBase.OKEnabled(True)
+        Else
+            ucrBase.OKEnabled(False)
+        End If
+    End Sub
 
-    Private Sub ucrSingle_Leave(sender As Object, e As EventArgs) Handles ucrReceiverNames.Leave
+    Private Sub ucrSelectVariables_DataFrameChanged() Handles ucrSelectVariables.DataFrameChanged
         ucrBase.clsRsyntax.AddParameter("data_name", Chr(34) & frmEditor.grdData.CurrentWorksheet.Name & Chr(34))
```
Comment (dannyparsons) [inline instat/dlgName.vb]: When the `DataFrameChanged` event is raised, you should change the parameter to the value in the data frame selector. It shouldn't always be the current worksheet.

File: instat/dlgName.vb [hunk 5]
```diff
@@ -16,53 +16,61 @@
...
-        ucrBase.clsRsyntax.AddParameter("column_name", Chr(34) & frmEditor.grdData.CurrentWorksheet.ColumnHeaders(frmEditor.grdData.CurrentWorksheet.SelectionRange.Col).[Text] & Chr(34))
+        TestOKEnabled()
     End Sub
 
-    Private Sub txtName_Leave(sender As Object, e As EventArgs) Handles txtName.Leave
-        ucrBase.clsRsyntax.AddParameter("new_val", Chr(34) & txtName.Text & Chr(34))
```
Comment (dannyparsons) [inline instat/dlgName.vb]: Why did you remove this line from here? Now you are not using the text from `txtName`

File: instat/dlgName.vb [hunk 6]
```diff
@@ -16,53 +16,61 @@
...
+        SetDefaults()
+        TestOKEnabled()
     End Sub
 
-    Private Sub ucrSingle_ValueChanged(sender As Object, e As EventArgs) Handles ucrReceiverNames.ValueChanged
-        If bUseDefaultName Then
-            txtName.Text = ucrReceiverNames.txtReceiverSingle.Text
-        End If
+    Private Sub ucrReceiverName_SelectionChanged(sender As Object, e As EventArgs) Handles ucrReceiverName.SelectionChanged
+        ucrBase.clsRsyntax.AddParameter("column_name", Chr(34) & ucrReceiverName.txtReceiverSingle.Text & Chr(34))
```
Comment (dannyparsons) [inline instat/dlgName.vb]: This is OK and works, but you should be using `GetVariableNames` to extract the selected items from the receivers, so that we always do this in a consistent way.

**Comment by deaspo** [comment]
The Reset() for the selector does not play nice if I merge the SetSettings and SetDefaults. Even if I merge the two, it always reset the receiver and txtName back to blanks even after passing in the optional parameters. I can move it to the click ClickReset from the setDefaults @dannyparsons

**Comment by deaspo** [comment]
And also bFirstLoad, as long as it is true, always uses the default options of ""

**Comment by volloholic** [comment]
Hi Polycarp
One way to solve this is to store the default values on the form and then use those values for the reset. 
I do not understand what you mean about the problem with bFirstLoad? I assume this code runs before you recall set defaults externally from the right click so this will work fine. 
Cheers
David

**Comment by deaspo** [comment]
Hi David,
The code for `If bFirstLoad Then
            SetDefaults()
            bFirstLoad = False
        End If`
runs after I have called setDefaults externally from the right click. So it over writes the values I had passed to the set defaults. This only happens when opening the dialog for the first time.

**Comment by volloholic** [comment]
That is strange it should run before! 
This problem can be resolved by moving the set bFirstLoad=False into set defaults.

**Comment by deaspo** [comment]
Still modifying the rename column

**Comment by deaspo** [comment]
Hi @dannyparsons please have a look at the code and give your comments. Tried the suggested option and were not wotking for the first load but were working for the subsequent loading of the form.

**Comment by dannyparsons** [comment]
Ok thanks, I'll have a look at this.

File: instat/dlgName.vb [hunk 7]
```diff
@@ -16,54 +16,65 @@
...
-        ucrBase.clsRsyntax.AddParameter("data_name", Chr(34) & frmEditor.grdData.CurrentWorksheet.Name & Chr(34))
-        ucrBase.clsRsyntax.AddParameter("column_name", Chr(34) & frmEditor.grdData.CurrentWorksheet.ColumnHeaders(frmEditor.grdData.CurrentWorksheet.SelectionRange.Col).[Text] & Chr(34))
+    Private Sub ucrSelectVariables_DataFrameChanged() Handles ucrSelectVariables.DataFrameChanged
+        ucrBase.clsRsyntax.AddParameter("data_name", ucrSelectVariables.ucrAvailableDataFrames.cboAvailableDataFrames.SelectedItem)
     End Sub
 
-    Private Sub txtName_Leave(sender As Object, e As EventArgs) Handles txtName.Leave
-        ucrBase.clsRsyntax.AddParameter("new_val", Chr(34) & txtName.Text & Chr(34))
+    Private Sub txtName_KeyPress(sender As Object, e As KeyPressEventArgs) Handles txtName.KeyPress
+        TestOKEnabled()
```
Comment (dannyparsons) [inline instat/dlgName.vb]: Not sure why you want this hear. You can do `TestOKEnabled()` under the `Leave` event instead.

**Comment by dannyparsons** [comment]
Did you try David's suggestion of moving `bFirstLoad=False` into set defaults? I think that's how we can get it to work in both cases.

**Comment by deaspo** [comment]
I did try but again all your suggestions were working on subsequent loads and not the first load

**Comment by dannyparsons** [comment]
Ok so would it be helpful if I played around with the code in a branch to
look at the loading issue now?

**Comment by deaspo** [comment]
Sure
